### PR TITLE
Masks for non-pilatus-like detectors

### PR DESCRIPTION
**Pull request by dagewa**
Rather than using -2 to indicate masked pixels for all detectors, set explicitly-masked pixels to `MASK_VAL`. This fixes the problem with valid -2 values being shown as masked in the image viewer (#536). This branch also requires https://github.com/cctbx/cctbx_project/tree/non-pilatus-flags

### CODE REVIEW

File: util/image_viewer/spotfinder_frame.py [hunk 1]
```diff
@@ -639,7 +639,9 @@ def draw_resolution_rings(self, unit_cell=None, space_group=None):
         # FIXME Currently assuming that all panels are in same plane
         p_id = detector.get_panel_intersection(beam.get_s0())
         if p_id == -1:
-            p_id = 0  # XXX beam doesn't intersect with any panels - is there a better solution?
+            p_id = (
+                0
+            )  # XXX beam doesn't intersect with any panels - is there a better solution?
```
Comment (rjgildea) [inline util/image_viewer/spotfinder_frame.py]: Would this comment be better moved to the line above to avoid this ugly formatting?

Comment (dagewa) [reply util/image_viewer/spotfinder_frame.py]: Probably. Black did this (I didn't touch the line on purpose)

**Comment by dagewa** [comment]
@phyy-nx I'm keen to get this in to DIALS 2.1. Might you be able to check things are okay over your side? In particular, there is a small and I think safe change in https://github.com/cctbx/cctbx_project/tree/non-pilatus-flags, but I don't know if that has any effects outside dials.image_viewer usage.

**Comment by dagewa** [comment]
Thanks!

**Pull request merged**